pkg/ottl/expression.go

```diff
@@ -180,23 +181,37 @@ type mapGetter[K any] struct {
 }
 
 func (m *mapGetter[K]) Get(ctx context.Context, tCtx K) (any, error) {
-	evaluated := map[string]any{}
+	result := pcommon.NewMap()
 	for k, v := range m.mapValues {
 		val, err := v.Get(ctx, tCtx)
 		if err != nil {
 			return nil, err
 		}
-		switch t := val.(type) {
+		switch typedVal := val.(type) {
 		case pcommon.Map:
-			evaluated[k] = t.AsRaw()
+			target := result.PutEmpty(k).SetEmptyMap()
+			typedVal.CopyTo(target)
+		case []any:
```

Comment:
Should we handle `pcommon.Slice` or other array types as well?

Reply:
This should not be required here, as internally the `newGetter` will return a `listGetter` for list items, which then returns `[]any`

Reply:
ok seems like there is another nuance to this:  a slice containing map entries will then return a `[]any`, but with `pcommon.Map` entries in it.
Should we maybe reconsider the original idea of returning `map[string]any` in the `mapGetter`, as opposed to `pcommon.Map`? This might be the most straight forward way of achieving consistency, especially because also the `listGetter` returns `[]any`, as opposed to `pcommon.Slice`. @edmocosta FYI

Reply:
I think this case is now also solved. I remembered we still have to use `pcommon.Map` as a return type of the `mapGetter` , as otherwise statements like `set(attributes, {"foo":"bar})` do not work. However the case I mentioned above can be solved in the `ValueExpression` getter. Now all the tests are passing again, including the one mentioned above. I will see if I can add some more test cases though.